diff --git a/factory/Makefile.am b/factory/Makefile.am
index bc6da8e73..04deffbbe 100644
--- a/factory/Makefile.am
+++ b/factory/Makefile.am
@@ -1,4 +1,4 @@
-ACLOCAL_AMFLAGS = -I ../m4
+ACLOCAL_AMFLAGS = -I m4
 
 SUBDIRS=include/factory
 
diff --git a/factory/configure.ac b/factory/configure.ac
index cf776e762..abe6524ec 100644
--- a/factory/configure.ac
+++ b/factory/configure.ac
@@ -14,8 +14,8 @@ dnl #
 #
 AC_INIT([factory], [4.4.1])
 AC_CONFIG_SRCDIR(canonicalform.cc)
-AC_CONFIG_MACRO_DIR([../m4])
-AC_CONFIG_AUX_DIR([../build-aux])
+AC_CONFIG_MACRO_DIR([m4])
+AC_CONFIG_AUX_DIR([build-aux])
 AC_CONFIG_HEADERS([_config.h])
 
 AM_INIT_AUTOMAKE([-Wall foreign subdir-objects]) # -Wno-extra-portability -Werror silent-rules
@@ -97,7 +97,7 @@ AC_ARG_WITH(
   Singular,
   [AS_HELP_STRING([--without-Singular],[(obsolete) build NOT for the use with CAS Singular.])],
   ,
-  [with_Singular=yes])
+  [with_Singular=no])
 
 PKG_REQUIRE="$PKG_REQUIRE"
 AC_SUBST(PKG_REQUIRE)
