Description: Avoid a nauty bug.
Bug: https://github.com/Macaulay2/M2/issues/1408
Author: Doug Torrance <dtorrance@piedmont.edu>
Last-Update: 2020-07-20

--- a/M2/Macaulay2/packages/Nauty.m2
+++ b/M2/Macaulay2/packages/Nauty.m2
@@ -1495,7 +1495,7 @@
         Example
             R = QQ[a..f];
             G = graph flatten apply({a,c,e}, v->v*{b,d,f})
-            relabelBipartite G
+            -- relabelBipartite G (skipping because of a nauty bug)
         Text
             If any of the inputs are not bipartite graphs, then the method
             throws an error.
