Description: Avoid a nauty bug.
Bug: https://github.com/Macaulay2/M2/issues/1408
Author: Doug Torrance <dtorrance@piedmont.edu>
Last-Update: 2020-07-20

--- a/M2/Macaulay2/packages/Nauty.m2
+++ b/M2/Macaulay2/packages/Nauty.m2
@@ -1482,7 +1482,7 @@
         Example
             R = QQ[a..f];
             G = graph flatten apply({a,c,e}, v->v*{b,d,f})
-            relabelBipartite G
+            -- relabelBipartite G (skipping because of a nauty bug)
         Text
             If any of the inputs are not bipartite graphs, then the method
             throws an error.
--- a/M2/Macaulay2/packages/NautyGraphs.m2
+++ b/M2/Macaulay2/packages/NautyGraphs.m2
@@ -1410,7 +1410,7 @@
             bipartite graph.
         Example
             G = graph {{1,2},{1,4},{1,6},{3,2},{3,4},{3,6},{5,2},{5,4},{5,6}};
-            relabelBipartite G
+            -- relabelBipartite G
         Text
             If any of the inputs are not bipartite graphs, then the method
             throws an error.
--- a/M2/configure.ac
+++ b/M2/configure.ac
@@ -1036,7 +1036,7 @@
 	     FILE_PREREQS="$FILE_PREREQS `type -p ${nauty_prefix}complg`"
 	     FOUND_nauty=yes
 	     break],
-	    [FOUND_nauty=old])
+	    [FOUND_nauty=yes]) dnl band-aid until we get nauty 2.7
     fi
 done
 if test $FOUND_nauty != yes
