Description: Investigate big endian build failures.
Author: Doug Torrance <dtorrance@piedmont.edu>
Bug: https://github.com/Macaulay2/M2/issues/2162
Last-Update: 2021-07-04

--- a/M2/Macaulay2/packages/EngineTests/GB.Test.Mathic.m2
+++ b/M2/Macaulay2/packages/EngineTests/GB.Test.Mathic.m2
@@ -170,7 +170,7 @@
   time g3 = groebnerBasis(ideal J1_*, Strategy=>"F4");
   time g1 = gens gb ideal(J1_*);
   assert(g1 == g2)
-  assert(g1 == g3)
+  assert Equation(g1, g3)
 
   R1 = ZZ/32003[w,x,y,z,MonomialOrder => {1,1,1,1}] -- this one is much slower for F4, seems
     -- TODO: to give more complicated GB elements?? why?
