Description: Investigate big endian build failures.
Author: Doug Torrance <dtorrance@piedmont.edu>
Bug: https://github.com/Macaulay2/M2/issues/2162
Last-Update: 2021-07-04

--- a/M2/Macaulay2/packages/EngineTests/GB.Test.Mathic.m2
+++ b/M2/Macaulay2/packages/EngineTests/GB.Test.Mathic.m2
@@ -184,7 +184,7 @@
   --time groebnerBasis(J1, Strategy=>"F4", "MGBOptions"=>{"Log"=>"all"})
   time g1 = gens gb (ideal J1_*);
   assert(g1 == g2)
-  assert(g1 == g3)
+  assert Equation(g1, g3)
   
   R1 = ZZ/32003[w,x,y,z,MonomialOrder => Eliminate 3]
   J1 = sub(J1, R1)
