Description: Fix mathicgb on big-endian architectures.
Author: Jerry James <loganjerry@gmail.com>
Origin: https://github.com/Macaulay2/mathicgb/pull/9/commits/b19bc2f
Bug: https://github.com/Macaulay2/mathicgb/issues/3
Bug: https://github.com/Macaulay2/mathicgb/issues/11
Bug: https://github.com/Macaulay2/M2/issues/2162
Last-Update: 2021-06-28

--- a/M2/Macaulay2/e/mathicgb/MonoMonoid.hpp
+++ b/M2/Macaulay2/e/mathicgb/MonoMonoid.hpp
@@ -582,13 +582,9 @@
       MATHICGB_ASSERT(i == varCount() / 2 || access(a, i*2+1) >= 0);
       
       uint64 A, B, AB;
-      // We have to use std::memcpy here because just casting to a int64 breaks
-      // the strict aliasing rule which implies undefined behavior. Both MSVC and
-      // gcc don't actually call memcpy here. MSVC is a tiny bit slower for this
-      // code than for casting while GCC seems to be exactly the same speed.
-      std::memcpy(&A, ptr(a, i*2), 8);
-      std::memcpy(&B, ptr(b, i*2), 8);
-      std::memcpy(&AB, ptr(ab, i*2), 8);
+      A = *ptr(a, i*2) | (((uint64)*ptr(a, i*2 + 1)) << 32);
+      B = *ptr(b, i*2) | (((uint64)*ptr(b, i*2 + 1)) << 32);
+      AB = *ptr(ab, i*2) | (((uint64)*ptr(ab, i*2 + 1)) << 32);
       orOfXor |= AB ^ (A + B);
     }
     MATHICGB_ASSERT((orOfXor == 0) == isProductOf(a, b, ab));
@@ -608,11 +604,11 @@
     uint64 orOfXor = 0;
     for (VarIndex i = varCount() / 2; i != beforeEntriesIndexBegin(); --i) {
       uint64 A1, A2, B, A1B, A2B;
-      std::memcpy(&A1, ptr(a1, i*2), 8);
-      std::memcpy(&A2, ptr(a2, i*2), 8);
-      std::memcpy(&B, ptr(b, i*2), 8);
-      std::memcpy(&A1B, ptr(a1b, i*2), 8);
-      std::memcpy(&A2B, ptr(a2b, i*2), 8);
+      A1 = *ptr(a1, i*2) | (((uint64)*ptr(a1, i*2 + 1)) << 32);
+      A2 = *ptr(a2, i*2) | (((uint64)*ptr(a2, i*2 + 1)) << 32);
+      B = *ptr(b, i*2) | (((uint64)*ptr(b, i*2 + 1)) << 32);
+      A1B = *ptr(a1b, i*2) | (((uint64)*ptr(a1b, i*2 + 1)) << 32);
+      A2B = *ptr(a2b, i*2) | (((uint64)*ptr(a2b, i*2 + 1)) << 32);
       orOfXor |= (A1B ^ (A1 + B)) | (A2B ^ (A2 + B));
     }
     MATHICGB_ASSERT
