Description: Replace build paths in documentation
Author: Doug Torrance <dtorrance@piedmont.edu>
Last-Update: 2020-07-15

--- a/M2/Macaulay2/m2/installPackage.m2
+++ b/M2/Macaulay2/m2/installPackage.m2
@@ -443,7 +443,7 @@
      if any({srcdir, builddir, homedir}, dir -> match(dir, outstr))
      then (
 	 -- .m2 files in source directory
-	 outstr = replace(srcdir | "Macaulay2/m2",
+	 outstr = replace(srcdir | "Macaulay2/\\b(m2|Core)\\b",
 	     finalPrefix | Layout#1#"packages" | "Core", outstr);
 	 outstr = replace(srcdir | "Macaulay2/packages/",
 	     finalPrefix | Layout#1#"packages", outstr);
@@ -455,6 +455,8 @@
 	     outstr = replace(prefixdir | Layout#2#key,
 		 finalPrefix | Layout#1#key, outstr));
 	 outstr = replace(prefixdir, finalPrefix, outstr);
+	 -- usr-build/bin is in PATH during build
+	 outstr = replace(builddir | "usr-build/", finalPrefix, outstr);
 	 -- home directory
 	 outstr = replace(homedir, "/home/m2user", outstr);
 	 );
--- a/M2/Macaulay2/packages/PushForward.m2
+++ b/M2/Macaulay2/packages/PushForward.m2
@@ -184,7 +184,7 @@
   A = kk[a,d]
   use R
   F = map(R,A)
-  pushFwd F
+  netList toList pushFwd F
   ///,
   TEX "In a previous version of this package, the third output was a function which assigned to each element of the target of ", TT "f", " its representation as an element of a free module which surjected onto the pushed forward module."
   }
