Description: Check if we can get frobby's version number.
Author: Doug Torrance <dtorrance@piedmont.edu>
Last-Update: 2016-03-25

--- a/M2/Macaulay2/d/version.dd
+++ b/M2/Macaulay2/d/version.dd
@@ -38,8 +38,12 @@
    extern char timestamp[];
    #undef ASSERT
    #ifdef HAVE_FROBBY
-     #include <stdinc.h>		// to get version number of frobby
-     #define FROBBY_VERSION constants::version
+     #if HAVE_FROBBY_VERSION_NUMBER
+       #include <stdinc.h>		// to get version number of frobby
+       #define FROBBY_VERSION constants::version
+     #else
+       #define FROBBY_VERSION \"\"
+     #endif
    #else
      #define FROBBY_VERSION \"not present\"
    #endif
--- a/M2/configure.ac
+++ b/M2/configure.ac
@@ -1062,6 +1062,20 @@
 		 BUILD_frobby=yes
 		 AC_MSG_RESULT([no, will build it])])
      fi
+     AC_DEFINE([HAVE_FROBBY_VERSION_NUMBER], [1],
+	 [if we can get version number from frobby])
+     if test $BUILD_frobby = no
+     then
+	AC_MSG_CHECKING([if we can get version number from frobby])
+	AC_LANG([C++])
+	AC_COMPILE_IFELSE(
+	    [AC_LANG_PROGRAM(
+		[#include <stdinc.h>],
+		[constants::version])],
+	    [AC_MSG_RESULT([yes])],
+	    [AC_MSG_RESULT([no])
+	     AC_DEFINE([HAVE_FROBBY_VERSION_NUMBER], [0])])
+     fi
      test $BUILD_frobby = yes && BUILTLIBS="-lfrobby $BUILTLIBS"
 fi
 
