--- a/M2/Macaulay2/packages/Nauty.m2
+++ b/M2/Macaulay2/packages/Nauty.m2
@@ -472,6 +472,8 @@
     scan(graphToString \ dataList, d -> o << d << endl);
     close o;
     -- try to harvest the lines
+    print(cmdStr | " < " | infn);
+    print(get infn);
     r := lines try get openIn ("!" | nauty'path | cmdStr | " <" | infn | " 2>" | erfn)
     else (
         -- nauty errored, harvest the error
