--- a/M2/check-configure/Makefile.in
+++ b/M2/check-configure/Makefile.in
@@ -29,6 +29,7 @@
 	    --with-system-memtailor=$(SYSTEM_MEMTAILOR)				\
 	    --with-system-mathic=$(SYSTEM_MATHIC)				\
 	    --with-system-mathicgb=$(SYSTEM_MATHICGB)				\
+	    --enable-download=@DOWNLOAD@					\
 	    --disable-building
 distclean:: clean; rm -f Makefile
 Makefile: Makefile.in ; cd .. && ./config.status check-configure/Makefile
