Description: Don't build gfan; we use the Debian package.
Author: Doug Torrance <dtorrance@piedmont.edu>
Last-Update: 2020-05-07

--- a/M2/configure.ac
+++ b/M2/configure.ac
@@ -935,11 +935,7 @@
 
 AC_MSG_CHECKING(whether the package gfan is installed)
 if test "`type -t gfan`" = file
-then AC_MSG_RESULT([yes, will build anyway])
-     dnl currently the gfanInterface package doesn't know how to find gfan if we don't build it.
-     dnl see https://github.com/Macaulay2/M2/pull/1123
-     BUILD_gfan=yes
-     BUILD_ALWAYS="$BUILD_ALWAYS gfan"
+then AC_MSG_RESULT([yes])
 else AC_MSG_RESULT([no, will build])
      BUILD_gfan=yes
 fi
