--- a/M2/configure.ac
+++ b/M2/configure.ac
@@ -1188,9 +1188,7 @@
 
 # frobby depends on gmp; we build mpir and use it to replace gmp; therefore we must build frobby and
 # not try to link with one we find.  See https://github.com/Macaulay2/M2/issues/654
-BUILD_frobby=yes
-BUILD_ALWAYS="frobby $BUILD_ALWAYS"
-
+BUILD_frobby = no
 if test "$FROBBY" = yes
 then if test $BUILD_frobby = no
      then AC_MSG_CHECKING(for frobby library)
