Description: Don't build flint; use Debian package.
Author: Doug Torrance <dtorrance@piedmont.edu>
Last-Update: 2020-05-06

--- a/M2/configure.ac
+++ b/M2/configure.ac
@@ -906,11 +906,6 @@
 
 dnl factory needs to know:
 AC_SUBST(BUILD_flint)
-dnl  We always build flint, until flint comes out with a release whose version number we can check for.
-dnl  (Now we use flint for factoring integers, and only a recent development version does that well.)
-BUILD_flint=yes
-BUILD_ALWAYS="$BUILD_ALWAYS flint"
-dnl 
 if test $BUILD_flint = no
 then AC_LANG(C)
      AC_CHECK_HEADER(flint/flint.h,LIBS="-lflint -lgmp $LIBS",BUILD_flint=yes)
