--- a/M2/configure.ac
+++ b/M2/configure.ac
@@ -932,8 +932,7 @@
 # Another possible problem can arise if mpfr comes dynamically linked to libgmp, but we want
 # to use mpir instead.
 
-BUILD_mpc=yes
-BUILD_ALWAYS="mpc $BUILD_ALWAYS"
+BUILD_mpc=no
 # We always build mpc because it depends on mpfr, which we always build.
 # If we don't build mpc, then we may end up linking with the system's shared mpc library, which
 # would bring in the systems shared mpfr library, conflicting with our mpfr library, which
