Description: Don't build normaliz
Author: Doug Torrance <dtorrance@piedmont.edu>
Last-Update: 2020-05-16

--- a/M2/configure.ac
+++ b/M2/configure.ac
@@ -951,11 +951,6 @@
      BUILD_csdp=yes
 fi
 
-dnl we can't use the test below for the presence of normaliz until the package Normaliz knows
-dnl how to find normaliz on the path, and until our *.dmg and
-dnl *.deb making procedures know how to include it or set the dependencies appropriately
-BUILD_normaliz=yes
-BUILD_ALWAYS="$BUILD_ALWAYS normaliz"
 dnl AC_MSG_CHECKING(whether the package normaliz is installed)
 dnl if test "`type -t normaliz`" = file
 dnl then AC_MSG_RESULT(yes)
