--- a/M2/configure.ac
+++ b/M2/configure.ac
@@ -945,8 +945,7 @@
 #    Command exited with non-zero status 1
 # Actually, we actually don't use any code from mpc, just one struct definition!  So we could eliminate its use.
 
-BUILD_glpk=yes
-BUILD_ALWAYS="glpk $BUILD_ALWAYS"
+BUILD_glpk=no
 # We always build glpk because it depends on mpir (as gmp), which we always build, and we don't want to
 # have the static version of mpir in memory along with the dynamic version of gmp.
 
