Description: Don't build 4ti2
Author: Doug Torrance <dtorrance@piedmont.edu>
Last-Update: 2020-05-16

--- a/M2/configure.ac
+++ b/M2/configure.ac
@@ -936,10 +936,6 @@
 then BUILTLIBS="-lflint $BUILTLIBS -lm"
 fi
 
-dnl we can't use the test below for the presence of 4ti2 until the package FourTiTwo knows
-dnl how to find programs such as "markov" under the name "4ti2-markov", and until our *.dmg and
-dnl *.deb making procedures know how to include them or set the dependencies appropriately
-BUILD_4ti2=yes
 dnl AC_MSG_CHECKING(whether the package 4ti2 is installed)
 dnl if test "`type -t 4ti2-circuits`" = file
 dnl then AC_MSG_RESULT(yes)
