Description: Improve eigen.cpp compilation time/RAM.
Author: Anton Leykin <anton.leykin@gmail.com>
Origin: https://github.com/Macaulay2/M2/pull/2236
Last-Update: 2021-09-17

--- a/M2/Macaulay2/e/eigen.cpp
+++ b/M2/Macaulay2/e/eigen.cpp
@@ -1,3 +1,10 @@
+//#define NO_LAPACK
+/*
+Uncommenting the above has two consequences:
+  (1) This effectively eliminates the use of LAPACK (eigen is used instead for machine precision).
+  (2) This slows down the compilation (at the moment) due to heavy templating in eigen. 
+*/
+
 #include <cstdlib>
 
 #include <M2/math-include.h>
@@ -17,12 +24,14 @@
 using Complex = std::complex<Real>;
 using MatrixXmpRRR = Eigen::Matrix<Real,Eigen::Dynamic,Eigen::Dynamic>;
 using MatrixXmpCCC = Eigen::Matrix<Complex,Eigen::Dynamic,Eigen::Dynamic>;
+#ifdef NO_LAPACK
 using MatrixXmpRR = Eigen::Matrix<double,Eigen::Dynamic,Eigen::Dynamic>;
 using MatrixXmpCC = Eigen::Matrix<std::complex<double>,Eigen::Dynamic,Eigen::Dynamic>;
-
+#endif
 
 namespace EigenM2 {
 
+#ifdef NO_LAPACK  
 // RR/CC
 
 // Need to rewrite matrix conversion functions
@@ -262,6 +271,7 @@
 
   return true;
 }
+#endif
 
 // RRR/CCC
 
