--- a/M2/Macaulay2/e/dmat.cpp
+++ b/M2/Macaulay2/e/dmat.cpp
@@ -101,9 +101,10 @@
   else
     {
       DMatZZpFFPACK N(mat);
+      ZZpFFPACK::ElementType det;
       result_det = FFPACK::Det(mat.ring().field(),
+                               det,
                                mat.numRows(),
-                               mat.numColumns(),
                                N.array(),
                                mat.numColumns());
     }
