Description: Find the singular gftables directory.
Author: Doug Torrance <dtorrance@monmouthcollege.edu>
Last-Update: 2015-06-10

--- a/M2/Macaulay2/d/startup.m2.in
+++ b/M2/Macaulay2/d/startup.m2.in
@@ -284,7 +284,7 @@
 	       "programs" => "@programsdir@/",
 	       "program licenses" => "@licensesdir@/",
 	       "package" => "@packagesdir@/PKG/",
-	       "factory gftables" =>  "@packagesdir@/Core/factory/",
+	       "factory gftables" =>  "/usr/share/singular/factory/",
 	       "packagedoc" => "@docdir@/PKG/",
 	       "packageimages" => "@docdir@/PKG/images/",
 	       "packagetests" => "@docdir@/PKG/tests/",
--- a/M2/Macaulay2/m2/factor.m2
+++ b/M2/Macaulay2/m2/factor.m2
@@ -51,7 +51,7 @@
 -- We mimic the procedure for finding a finite field addition table used in the routine gf_get_table
 -- for building the file name in "gffilename", in the file BUILD_DIR/libraries/factory/build/factory/gfops.cc .
 -- Reminder: the contents of currentLayout are determined by the file ../d/startup.m2.in .
-gfdir = prefixDirectory | currentLayout#"factory gftables"
+gfdir = currentLayout#"factory gftables"
 gftestfile = gfdir | "gftables/961" -- 961==31^2
 if not fileExists gftestfile
 then error ("sample Factory finite field addition table file missing, needed for factorization: ", gftestfile)
