--- a/M2/Macaulay2/e/util.cpp
+++ b/M2/Macaulay2/e/util.cpp
@@ -10,7 +10,7 @@
   for (size_t i = 0; i < strs->len; i++)
     {
       M2_string a = strs->array[i];
-      std::string b(a->array, a->len);
+      std::string b(reinterpret_cast<char*>(a->array), a->len);
       result.push_back(b);
     }
 }
@@ -21,7 +21,7 @@
   for (size_t i = 0; i < strs->len; i++)
     {
       M2_string a = strs->array[i];
-      result.emplace_back(a->array, a->len);
+      result.emplace_back(reinterpret_cast<char*>(a->array), a->len);
     }
   return result;
 }
