Description: Fix i386 build errors
Author: Doug Torrance <dtorrance@piedmont.edu>
Origin: https://github.com/Macaulay2/M2/pull/1537
Bug: https://github.com/Macaulay2/M2/issues/1423
Bug: https://github.com/Macaulay2/M2/issues/1429
Last-Update: 2020-10-08

--- a/M2/Macaulay2/packages/IntegralClosure.m2
+++ b/M2/Macaulay2/packages/IntegralClosure.m2
@@ -1343,40 +1343,40 @@
      to instead compute Hom(J^-1, J^-1).  This is usually a more time consuming
      computation, but it does potentially get to the answer in a smaller number of steps.
    Example
-     S = QQ[x,y]
-     f = ideal (y^4-2*x^3*y^2-4*x^5*y+x^6-x^7)
+     S = QQ[x,y,z]
+     f = ideal (x^8-z^6-y^2*z^4-z^3)
      R = S/f
      time R' = integralClosure R
      netList (ideal R')_*
      icFractions R
    Example
-     S = QQ[x,y]
-     f = ideal (y^4-2*x^3*y^2-4*x^5*y+x^6-x^7)
+     S = QQ[x,y,z]
+     f = ideal (x^8-z^6-y^2*z^4-z^3)
      R = S/f
      time R' = integralClosure(R, Strategy => Radical)
      netList (ideal R')_*
      icFractions R
    Example
-     S = QQ[x,y]
-     f = ideal (y^4-2*x^3*y^2-4*x^5*y+x^6-x^7)
+     S = QQ[x,y,z]
+     f = ideal (x^8-z^6-y^2*z^4-z^3)
      R = S/f
      time R' = integralClosure(R, Strategy => AllCodimensions)
      netList (ideal R')_*
    Example
-     S = QQ[x,y]
-     f = ideal (y^4-2*x^3*y^2-4*x^5*y+x^6-x^7)
+     S = QQ[x,y,z]
+     f = ideal (x^8-z^6-y^2*z^4-z^3)
      R = S/f
      time R' = integralClosure(R, Strategy => SimplifyFractions)
      netList (ideal R')_*     
    Example
-     S = QQ[x,y]
-     f = ideal (y^4-2*x^3*y^2-4*x^5*y+x^6-x^7)
+     S = QQ[x,y,z]
+     f = ideal (x^8-z^6-y^2*z^4-z^3)
      R = S/f
      time R' = integralClosure (R, Strategy => RadicalCodim1)
      netList (ideal R')_*
    Example
-     S = QQ[x,y]
-     f = ideal (y^4-2*x^3*y^2-4*x^5*y+x^6-x^7)
+     S = QQ[x,y,z]
+     f = ideal (x^8-z^6-y^2*z^4-z^3)
      R = S/f
      time R' = integralClosure (R, Strategy => Vasconcelos)
      netList (ideal R')_*
--- a/M2/Macaulay2/packages/AlgebraicSplines.m2
+++ b/M2/Macaulay2/packages/AlgebraicSplines.m2
@@ -2509,7 +2509,7 @@
 	    V = {{1, 0, 0}, {-1, 0, 0}, {0, 1, 0}, {0, -1, 0}, {0, 0, 1}, {0, 0, -1}, {-2, -2, -2}, {-2, 2, -2}, {2, 2, -2}, {2, -2, -2}, {-2, -2, 2}, {-2, 2, 2}, {2, 2, 2}, {2, -2, 2}};
 	    F = {{0, 1, 2, 3, 4, 5}, {0, 8, 9, 12, 13}, {1, 6, 7, 10, 11}, {2, 7, 8, 11, 12}, {3, 6, 9, 10, 13}, {4, 10, 11, 12, 13}, {5, 6, 7, 8, 9}, {0, 2, 8, 12}, {0, 3, 9, 13}, {0, 4, 12, 13}, {0, 5, 8, 9}, {1, 2, 7, 11}, {1, 3, 6, 10}, {1, 4, 10, 11}, {1, 5, 6, 7}, {2, 4, 11, 12}, {3, 4, 10, 13}, {3, 5, 6, 9}, {2, 5, 7, 8}, {0, 2, 4, 12}, {0, 2, 5, 8}, {0, 3, 4, 13}, {0, 3, 5, 9}, {1, 2, 4, 11}, {1, 2, 5, 7}, {1, 3, 4, 10}, {1, 3, 5, 6}};
 	    C = splineComplex(V,F,1);
-	    associatedPrimes annihilator HH_2 C
+	    HH_2 C -- supposed to be "associatedPrimes annihilator HH_2 C", but this results in a SIGSEGV on i386 (see https://github.com/Macaulay2/M2/issues/1429)
 	Text
 	    The above example showcases a fairly complex three dimensional polyhedral complex.
 	    It is a three-dimensional analog of the Morgan-Scot partition in the sense that
@@ -2737,4 +2737,4 @@
 --	    S = splines(V,F,E,1) -- splines in R^2 with smoothness 1
 --    SeeAlso
 --        splines
---///
\ No newline at end of file
+--///
