Description: Fix Visualize in Firefox (symlinks won't work)
Bug: https://bugzilla.mozilla.org/803999
Author: Doug Torrance <dtorrance@piedmont.edu>
Forwarded: https://github.com/Macaulay2/M2/pull/2124
Last-Update: 2021-05-31

--- a/M2/Macaulay2/packages/Visualize.m2
+++ b/M2/Macaulay2/packages/Visualize.m2
@@ -546,8 +546,13 @@
 	    );
 	);
 
-    for dir in existingDirs do removeFile concatenate(dst, dir);
-    for dir in dirs do symlinkFile(basePath | "Visualize/" | dir, dst | dir);
+    for dir in dirs do (
+	makeDirectory(dst | dir);
+	for file in readDirectory(basePath | "Visualize/" | dir) do (
+	    if not member(file, {".", ".."}) then (
+		copyFile(realpath(basePath | "Visualize/" | dir | "/" | file),
+		    dst | dir | "/" | file)))
+    );
 
     return "Created directories at "|dst;
 )
