Description: Update StatePolytope for modern gfan and polymake
Author: Doug Torrance <dtorrance@piedmont.edu>
Forwarded: https://github.com/Macaulay2/M2/pull/1373
Last-Update: 2020-09-18

--- a/M2/Macaulay2/packages/StatePolytope.m2
+++ b/M2/Macaulay2/packages/StatePolytope.m2
@@ -23,8 +23,6 @@
      --"hilbertPt", 
      --"statePolytopePoints", 
      --"statePolytopePoints", 
-     --"printHilbPt", 
-     --"createPolymakeInputFile",
      "statePolytope", --documented
      --"maxUGBDegree", 
      "isStable" --documented
@@ -96,28 +94,24 @@
      sum apply(m+1,k-> statePolytopePoints(k,I,initialIdealsList))
      )
 
-printHilbPt = (L,I) -> ( str:= "1";
-for j from 0 to (numgens ring(I)-1) do str = concatenate(str,concatenate(" ",toString lift(L_j,ZZ)));
-return str
-)
-
-createPolymakeInputFile = (statePolytopePointsList,I) -> (
-openOut "temporarypolymakefile.txt";
-polymakeinput :=  concatenate("_application polytope\nPOINTS",newline);
-for i from 0 to (#statePolytopePointsList - 1) do polymakeinput = concatenate(polymakeinput,concatenate(printHilbPt(statePolytopePointsList_i,I),newline));
-"temporarypolymakefile.txt" << polymakeinput << closeOut
-)
-
-polymakeToM2 = (st) -> (
-p := concatenate("VERTICES",concatenate(newline,"1 "));
-st = replace(p,"{{",st);
-p = concatenate(newline,"1 ");
-st = replace(p,"},{",st);
-st = replace(" ",", ",st);
-st = concatenate(st,"}}");
-return value st
-)
+runPolymake = args -> (
+    fn := temporaryFileName();
+    output := fn | ".out";
+    err := fn | ".err";
+    result := run("polymake --no-config " | args | " > " | output |
+	" 2> " | err);
+    if result != 0 then error("polymake return an error:\n" | get err);
+    get output
+)
+
+M2toPolymake = points ->
+    "'my $p = new Polytope(POINTS=>[" |
+	demark(", ", apply(points, point ->
+	    "[" | demark(", ", toString \ prepend(1, point)) | "]")) |
+	"]); print $p->VERTICES;'"
 
+polymakeToM2 = (st) ->
+    apply(lines st, line -> value \ drop(separate(" ", line), 1))
 
 maxUGBDegree = (L) -> (
      (max apply(#L, i -> max apply(#(L_i), j -> degree L_i_j)))_0
@@ -126,50 +119,33 @@
 statePolytope = method(
      TypicalValue => List
      )
-statePolytope(ZZ,Ideal) := (m,I) -> ( initialIdealsList := initialIdeals(I);
-createPolymakeInputFile(statePolytopePoints(m,I,initialIdealsList),I);
-polymakesession := "!polymake temporarypolymakefile.txt VERTICES";
-polymakesession << closeOut;
-st := get polymakesession;
-polymakeToM2(st)
+statePolytope(ZZ,Ideal) := (m,I) -> (
+    initialIdealsList := initialIdeals(I);
+    st := runPolymake M2toPolymake statePolytopePoints(m,I,initialIdealsList);
+    polymakeToM2(st)
 )
 
-statePolytope(Ideal) := (I) -> ( initialIdealsList := initialIdeals(I);
-createPolymakeInputFile(fullStatePolytopePoints(maxUGBDegree(initialIdealsList),I,initialIdealsList),I);
-polymakesession := "!polymake temporarypolymakefile.txt VERTICES";
-polymakesession << closeOut;
-st :=  get polymakesession;
-polymakeToM2(st)
+statePolytope(Ideal) := (I) -> (
+    initialIdealsList := initialIdeals(I);
+    st := runPolymake M2toPolymake fullStatePolytopePoints(
+	maxUGBDegree(initialIdealsList),I,initialIdealsList);
+    polymakeToM2(st)
 )
 
-
-
-
-
 isStable = method(
      TypicalValue => Boolean
      )
-isStable(ZZ,Ideal) := (m,I) -> (   
+isStable(ZZ,Ideal) := (m,I) -> (
      initialIdealsList := initialIdeals(I);
-createPolymakeInputFile(statePolytopePoints(m,I,initialIdealsList),I);
-polymakesession := "!polymake temporarypolymakefile.txt VERTICES";
-polymakesession << closeOut;
-st := get polymakesession;
-    str:= st;
-     barycenter := toString ((sum(statePolytopePoints(m,I,initialIdeals(I)))_0) / (numgens ring(I) ));
-     barycenterstring:= concatenate("POINTS",concatenate(newline,"1 "));
-     for i from 0 to (numgens(ring(I))-1) do barycenterstring = concatenate(barycenterstring, concatenate(barycenter," "));
-     str = replace("VERTICES",barycenterstring,str);
-     openOut "augmentedtemporarypolymakefile.txt";
-     "augmentedtemporarypolymakefile.txt" << "_application polytope\n" << str << closeOut;
-     polymakesession2 := "!polymake augmentedtemporarypolymakefile.txt VERTICES";
-     polymakesession2 << closeOut;
-     augmentedstatepolytope := get polymakesession2;
-     if set polymakeToM2(st) === set polymakeToM2(augmentedstatepolytope)   then true else false
-          )
-     
-
-
+     points := statePolytopePoints(m,I,initialIdealsList);
+     st := runPolymake M2toPolymake points;
+     n := numgens ring I;
+     barycenter := apply(n, i -> sum points_0 / n);
+     augmentedpoints := prepend(barycenter, points);
+     augmentedstatepolytope :=  runPolymake M2toPolymake augmentedpoints;
+     if set polymakeToM2(st) === set polymakeToM2(augmentedstatepolytope)
+	then true else false
+)
 
 beginDocumentation()
 document {    
@@ -205,8 +181,8 @@
 TEST ///
 R=QQ[a..d];
 I=ideal(a*c-b^2,a*d-b*c,b*d-c^2);
-L= {{b*d, a*d, a*c}, {c^2, a*d, a*c}, {c^2, b*c, a*c, a^2*d}, {c^2, b*c, b^3,
-     a*c}, {c^2, b*c, b^2}, {b*d, b^2, a*d}, {b*d, b*c, b^2, a*d^2}, {c^3, b*d,
+L= {{b*d, a*d, a*c}, {a*d, b*d, b^2}, {a*d^2, b*d, b*c, b^2}, {b*d, b^2, b*c,
+     c^3}, {a*d, a*c, c^2}, {a^2*d, a*c, c^2, b*c}, {a*c, c^2, b*c, b^3}, {c^2,
      b*c, b^2}};
 assert(set initialIdeals(I) === set L)
 ///
