Description: Fix failing rawFactor test
Bug: https://github.com/Macaulay2/M2/issues/1500
Author: Dan Grayson <dan@math.uiuc.edu>
Origin: upstream, https://github.com/Macaulay2/M2/commit/a44b11d
Last-Update: 2020-10-01

--- a/M2/Macaulay2/tests/normal/factor.m2
+++ b/M2/Macaulay2/tests/normal/factor.m2
@@ -56,6 +56,10 @@
      t === ((raw 1_R, raw (x^2+a), raw (x^2-a)), (1, 1, 1)) 
      or
      t === ((raw 1_R, raw (x^2-a), raw (x^2+a)), (1, 1, 1)) 
+     or
+     t === ((raw (x^2+a), raw (x^2-a)), (1, 1))
+     or
+     t === ((raw (x^2-a), raw (x^2+a)), (1, 1))
      )
 
 -- factoring a Laurent polynomial
