Description: Add "ForeignStructType ForeignObject" method.
 Fixes failing "RInterface" test.
Author: Doug Torrance <dtorrance@piedmont.edu>
Origin: https://github.com/Macaulay2/M2/pull/3034
Last-Update: 2023-12-11

--- a/M2/Macaulay2/packages/ForeignFunctions.m2
+++ b/M2/Macaulay2/packages/ForeignFunctions.m2
@@ -463,6 +463,8 @@
 
 ForeignStructType VisibleList := (T, x) -> new T from x
 ForeignStructType HashTable := (T, x) -> T apply(keys x, k -> k => x#k)
+ForeignStructType ForeignObject := lookup(
+    symbol SPACE, ForeignType, ForeignObject)
 
 isAtomic ForeignStructType := T -> T.Atomic
 
