Description: Use GNU C++17 extensions.
Author: Doug Torrance <dtorrance@piedmont.edu>
Origin: https://github.com/Macaulay2/M2/pull/1923
Bug: https://github.com/Macaulay2/M2/issues/1901
Bug: https://github.com/Macaulay2/M2/issues/1902
Last-Update: 2021-02-12

--- a/M2/Macaulay2/e/Makefile.in
+++ b/M2/Macaulay2/e/Makefile.in
@@ -9,7 +9,7 @@
 include @srcdir@/Makefile.files
 include Makefile.common
 CPPFLAGS := -I@srcdir@ $(CPPFLAGS) -I../d -I@srcdir@/../c -Wno-fatal-errors
-CXXFLAGS += -Wno-unknown-pragmas -std=c++17
+CXXFLAGS += -Wno-unknown-pragmas -std=gnu++17
 
 DOXYGEN_CONF_DIR :=$(SRCDIR)/doxygen-settings
 
