Description: Fix failing rawFactor test
Bug: https://github.com/Macaulay2/M2/issues/1500
Author: Doug Torrance <dtorrance@piedmont.edu>
Last-Update: 2020-09-18

--- a/M2/Macaulay2/tests/normal/factor.m2
+++ b/M2/Macaulay2/tests/normal/factor.m2
@@ -53,9 +53,9 @@
 debug Core
 t = rawFactor ( raw(x^4 - 2), raw(a^2 - 2) )
 assert(
-     t === ((raw 1_R, raw (x^2+a), raw (x^2-a)), (1, 1, 1)) 
+     t === ((raw (x^2+a), raw (x^2-a)), (1, 1))
      or
-     t === ((raw 1_R, raw (x^2-a), raw (x^2+a)), (1, 1, 1)) 
+     t === ((raw (x^2-a), raw (x^2+a)), (1, 1))
      )
 
 -- factoring a Laurent polynomial
