--- a/M2/Macaulay2/m2/html.m2
+++ b/M2/Macaulay2/m2/html.m2
@@ -720,7 +720,8 @@
 	  "^\\.svn$", 
 	  -- The package Style has a read-only file "Makefile", made from "Makefile.in", that doesn't need to be distributed.
 	  -- Better would be to fix copyDirectory so it manages to copy even when the target file is read-only
-	  "Makefile"
+	  "Makefile",
+	  "^\\.gitignore$"
 	  };
 
      if pkg === Core then (
