Description: Temporary patch to skip over gmp bug.

--- a/M2/Macaulay2/d/M2inits.c
+++ b/M2/Macaulay2/d/M2inits.c
@@ -62,7 +62,6 @@
     M2inits_firsttime = 0;
     if (__gmp_allocate_func != __gmp_default_allocate) {
       fprintf(stderr,"internal error: gmp memory allocation functions already set to non-default value\n");
-      exit(1);
     }
     initializeGMP_Cwrapper(); /* this calls factory's initializeGMP() in factory/initgmp.cc, which will call __gmp_set_memory_functions just once */
   }
