--- a/M2/Macaulay2/m2/html.m2
+++ b/M2/Macaulay2/m2/html.m2
@@ -472,7 +472,7 @@ runFile := (inf,inputhash,outf,tmpf,desc,pkg,announcechange,usermode,examplefile
      src := concatenate apply(srcdirs, d -> (" --srcdir ",format d));
      -- we specify --no-readline because the readline library catches SIGINT:
      args := "--silent --print-width 77 --stop --int --no-readline" | (if usermode then "" else " -q") | src | setseed | ldpkg;
-     env := "GC_MAXIMUM_HEAP_SIZE=400M ";
+     env := "HOME=/home/username GC_MAXIMUM_HEAP_SIZE=400M ";
      cmdname := commandLine#0;
      -- must convert a relative path to an absolute path so we can run the same M2 from another directory while
      -- running the examples:
--- a/M2/Macaulay2/d/Makefile.in
+++ b/M2/Macaulay2/d/Makefile.in
@@ -225,13 +225,17 @@
 	@(\
 	 cat @srcdir@/startup-header.h; \
 	 echo 'cached_file startupFile = {' ; \
-	     echo @abs_srcdir@/startup.m2.in | sed $(BSTRING) ; \
+	     echo /path/to/source/M2/Macaulay2/d/startup.m2.in | \
+		sed $(BSTRING) ; \
 	     echo  ',' ; \
 	     cat startup.m2 | sed $(SSTRING) ; \
 	 echo  '};' ; \
 	 echo 'cached_file testStrings[] = {' ; \
 	 for i in @abs_srcdir@/basictests/*.m2 ; \
-	 do echo '{' ; echo $$i | sed $(BSTRING) ; echo  ',' ; cat $$i | sed $(SSTRING) ; echo  '},' ; \
+	 do echo '{' ; \
+		echo "/path/to/source/M2/Macaulay2/d/`basename $$i`" | \
+		sed $(BSTRING) ; echo  ',' ; cat $$i | sed $(SSTRING) ; \
+		echo  '},' ; \
 	 done ; \
 	 echo '};' ; \
 	 cat @srcdir@/startup-trailer.h; \
--- a/M2/Macaulay2/m2/code.m2
+++ b/M2/Macaulay2/m2/code.m2
@@ -26,7 +26,7 @@
 	  if #file < stop then error("line number ",toString stop, " not found in file ", filename);
 	  while stop >= start and file#(stop-1) === "" do stop = stop-1;
 	  stack prepend(
-	       concatenate(filename, ":", 
+	       concatenate(reproduciblePath(filename), ":", 
 		    toString start, ":", toString (startcol+1),
 		    "-",
 		    toString stop, ":", toString (stopcol+1),
--- a/M2/Macaulay2/m2/nets.m2
+++ b/M2/Macaulay2/m2/nets.m2
@@ -66,7 +66,7 @@
      if hasAttribute(f,ReverseDictionary) then return toString getAttribute(f,ReverseDictionary);
      t := locate f;
      if t === null then "-*Function*-" 
-     else concatenate("-*Function[", t#0, ":", toString t#1| ":", toString (t#2+1), "-", toString t#3| ":", toString (t#4+1), "]*-")
+     else concatenate("-*Function[", reproduciblePath(t#0), ":", toString t#1| ":", toString (t#2+1), "-", toString t#3| ":", toString (t#4+1), "]*-")
      )
 
 net FunctionBody := toString FunctionBody := f -> (
--- a/M2/Macaulay2/m2/debugging.m2
+++ b/M2/Macaulay2/m2/debugging.m2
@@ -1,8 +1,22 @@
 --		Copyright 1993-2002 by Daniel R. Grayson
 
+reproduciblePath = buildPath -> (
+	-- m2 -> core
+	result := regex("/M2/Macaulay2/m2/", buildPath);
+	if result =!= null then
+		return "/usr/share/Macaulay2/Core/" |
+			substring(first first result + 17, buildPath);
+	-- packages
+	result = regex("/M2/Macaulay2/packages/", buildPath);
+	if result =!= null then
+		return "/usr/share/Macaulay2/" |
+			substring(first first result + 23, buildPath);
+	buildPath
+	)
+
 symbolLocation = s -> (
      t := locate s;
-     if t =!= null then t#0 | ":" | toString t#1| ":" | toString (t#2+1) | "-" | toString t#3| ":" | toString (t#4+1)
+     if t =!= null then reproduciblePath(t#0) | ":" | toString t#1| ":" | toString (t#2+1) | "-" | toString t#3| ":" | toString (t#4+1)
      else "")
 
 processArgs := args -> concatenate (
