Description: Various patches for reproducible builds.
Origin: https://github.com/d-torrance/M2/tree/reproducible-builds
Last-Update: 2020-05-24

--- a/M2/Macaulay2/d/Makefile.in
+++ b/M2/Macaulay2/d/Makefile.in
@@ -225,13 +225,17 @@
 	@(\
 	 cat @srcdir@/startup-header.h; \
 	 echo 'cached_file startupFile = {' ; \
-	     echo @abs_srcdir@/startup.m2.in | sed $(BSTRING) ; \
+	     echo /path/to/source/M2/Macaulay2/d/startup.m2.in | \
+		sed $(BSTRING) ; \
 	     echo  ',' ; \
 	     cat startup.m2 | sed $(SSTRING) ; \
 	 echo  '};' ; \
 	 echo 'cached_file testStrings[] = {' ; \
 	 for i in @abs_srcdir@/basictests/*.m2 ; \
-	 do echo '{' ; echo $$i | sed $(BSTRING) ; echo  ',' ; cat $$i | sed $(SSTRING) ; echo  '},' ; \
+	 do echo '{' ; \
+		echo "/path/to/source/M2/Macaulay2/d/`basename $$i`" | \
+		sed $(BSTRING) ; echo  ',' ; cat $$i | sed $(SSTRING) ; \
+		echo  '},' ; \
 	 done ; \
 	 echo '};' ; \
 	 cat @srcdir@/startup-trailer.h; \
