Description: Don't download gtest.
Author: Doug Torrance <dtorrance@piedmont.edu>
Last-Update: 2015-06-04

--- a/M2/configure.ac
+++ b/M2/configure.ac
@@ -788,7 +788,7 @@
 
 # a library is listed in BUILD_ALWAYS if and only if we *always* build it, 
 # even if the option "--disable-building" is specified:
-BUILD_ALWAYS=" gtest "
+BUILD_ALWAYS=""
 
 for i in $LIBLIST
 do eval BUILD_$i=no
@@ -842,7 +842,7 @@
     ENABLE_GIVARO=yes
     )
 
-BUILD_gtest=yes
+BUILD_gtest=no
 
 PROGLIST=" $PROGLIST "
 AC_ARG_WITH(unbuilt-programs, 
