Description: Skip Matroids areIsomorphic test; alarm sometimes not caught
Author: Doug Torrance <dtorrance@piedmont.edu>
Bug: https://github.com/Macaulay2/M2/issues/1392
Last-Update: 2020-11-26

--- a/M2/Macaulay2/packages/Matroids.m2
+++ b/M2/Macaulay2/packages/Matroids.m2
@@ -3645,7 +3645,6 @@
 F1 = set{0,1,2,3,7}
 F2 = F1 + set{5,8}
 assert(areIsomorphic(uniformMatroid(2,2), minor(M0, F1, M0.groundSet - F2)))
-assert(areIsomorphic(M0, matroid graph edges graph M0_*))
 Delta = independenceComplex M0
 F = fVector Delta
 assert(ideal Delta == ideal M0 and F === fVector independenceComplex M1)
