Description: Skip LLLBases test that segfaults on i386.
Author: Doug Torrance <dtorrance@piedmont.edu>
Bug: https://github.com/Macaulay2/M2/issues/1564
Last-Update: 2020-10-28

--- a/M2/Macaulay2/packages/LLLBases.m2
+++ b/M2/Macaulay2/packages/LLLBases.m2
@@ -1401,23 +1401,6 @@
   a = LLL m
   assert (a == m)
 ///
-TEST ///
-    setRandomSeed 0
-    m0 = random(ZZ^20, ZZ^30, Height=>100000)
-    m1 = syz m0;
-    m = m1;
-
-    setRandomSeed 0
-    m0 = random(ZZ^40, ZZ^50, Height=>10000)
-    m1 = syz m0;
-    m2 = kernelLLL m0;
-    m = m1;
-    m = m2;
-
-    time LLL(m1);
-    ---time LLL(m1, Strategy=>fpLLL);
-    -- TODO: run these on all variants that work
-///
 
 -------------------
 -- hermite -----
