Description: Skip failing tests
Author: Doug Torrance <dtorrance@piedmont.edu>
Bug: https://github.com/Macaulay2/M2/issues/1330
Last-Update: 2020-06-23

--- a/M2/Macaulay2/packages/Makefile.in
+++ b/M2/Macaulay2/packages/Makefile.in
@@ -37,7 +37,7 @@
 
 STOP = --stop --silent
 $(foreach i,\
-	$(sort $(ALL_PACKAGES) $(DEVEL)),\
+	$(sort $(filter-out RunExternalM2, $(ALL_PACKAGES) $(DEVEL))),\
 	$(eval check::check-$i)\
 	$(eval check-$i:; \
 		if ! grep "CacheExampleOutput => true" @srcdir@/$i.m2 >/dev/null ;\
