Description: Skip example requiring topcom
Author: Doug Torrance <dtorrance@piedmont.edu>
Last-Update: 2020-05-16

--- a/M2/Macaulay2/packages/Polyhedra/documentation/documentation.m2
+++ b/M2/Macaulay2/packages/Polyhedra/documentation/documentation.m2
@@ -120,9 +120,6 @@
          The lattice volume describes how many unit simplices fit into the polyhedron.
          It is computed via a triangulation of the polyhedron and then summing over the
          volumes of the simplices.
-
-      Example
-         latticeVolume hypercube 3
 ///
 
 doc ///
@@ -390,8 +387,6 @@
       Text
          This method computes a regular triangulation of a polytope using the package {\tt Topcom.m2}. The output is a list of list of indices which vertices of {\tt P} give a simplex in the triangulation.
 
-      Example
-         T = regularTriangulation hypercube 2
 ///
 
 doc ///
--- a/M2/Macaulay2/packages/Polyhedra/documentation/old_documentation.m2
+++ b/M2/Macaulay2/packages/Polyhedra/documentation/old_documentation.m2
@@ -1936,8 +1936,7 @@
      
      EXAMPLE {
 	  " P = crossPolytope 2",
-	  " Q = hypercube 2",
-	  " mixedVolume {P,Q}"
+	  " Q = hypercube 2"
 	  }
      
      }
@@ -2232,8 +2231,7 @@
      of a simplex is |det(v_1-v_0,..,v_n-v_0)|/n!, where v_0,..,v_n are the vertices of the simplex.",
      
      EXAMPLE {
-	  " P = crossPolytope 3",
-	  " volume P"
+	  " P = crossPolytope 3"
 	  }
      
      }
@@ -3040,9 +3038,7 @@
      ",TT "to be added",".",
      
      EXAMPLE {
-	  " P = crossPolytope 2",
-	  " Q = secondaryPolytope P",
-	  " vertices Q"
+	  " P = crossPolytope 2"
 	  }
      }
 
