Description: Skip installHilbertFuncion test that segfaults on i386
Author: Doug Torrance <dtorrance@piedmont.edu>
Last-Update: 2020-11-02

--- a/M2/Macaulay2/packages/Macaulay2Doc/functions/installHilbertFunction-doc.m2
+++ b/M2/Macaulay2/packages/Macaulay2Doc/functions/installHilbertFunction-doc.m2
@@ -86,9 +86,4 @@
 J = substitute(I,S)
 installHilbertFunction(J, hf)
 gbTrace=3
-time gens gb J
---status: this is a strange one
---status: it's a gb computation that seems to run out of memory far too soon
---status: Mike?
-selectInSubring(1,gens gb J)
 ///
