Description: Skip installHilbertFuncion test that segfaults on i386
Bug: https://github.com/Macaulay2/M2/issues/1577
Author: Doug Torrance <dtorrance@piedmont.edu>
Last-Update: 2020-11-03

--- a/M2/Macaulay2/tests/normal/core-tests.m2
+++ b/M2/Macaulay2/tests/normal/core-tests.m2
@@ -1878,11 +1878,6 @@
 J = substitute(I,S)
 installHilbertFunction(J, hf)
 gbTrace=3
-time gens gb J
---status: this is a strange one
---status: it's a gb computation that seems to run out of memory far too soon
---status: Mike?
-selectInSubring(1,gens gb J)
 ///
 
 TEST' ///
