Description: Skip DiffAlg test that sometimes aborts on armhf
Author: Doug Torrance <dtorrance@piedmont.edu>
Bug: https://github.com/Macaulay2/M2/issues/1728
Last-Update: 2020-12-23

--- a/M2/Macaulay2/packages/DiffAlg.m2
+++ b/M2/Macaulay2/packages/DiffAlg.m2
@@ -1316,6 +1316,7 @@
 ///
 
 TEST ///
+-* no-capture-flag *-
 w = random newForm(3,2,2,"a")
 h = newForm(3,1,1,"b")
 L = genKer(w^(diff h) + h ^ (diff w), h)
