Description: Skip things requiring topcom until it's in Debian
Author: Doug Torrance <dtorrance@piedmont.edu>
Last-Update: 2020-05-22

--- a/M2/Macaulay2/packages/Polyhedra/documentation/documentation.m2
+++ b/M2/Macaulay2/packages/Polyhedra/documentation/documentation.m2
@@ -120,9 +120,6 @@
          The lattice volume describes how many unit simplices fit into the polyhedron.
          It is computed via a triangulation of the polyhedron and then summing over the
          volumes of the simplices.
-
-      Example
-         latticeVolume hypercube 3
 ///
 
 doc ///
@@ -390,8 +387,6 @@
       Text
          This method computes a regular triangulation of a polytope using the package {\tt Topcom.m2}. The output is a list of list of indices which vertices of {\tt P} give a simplex in the triangulation.
 
-      Example
-         T = regularTriangulation hypercube 2
 ///
 
 doc ///
--- a/M2/Macaulay2/packages/Polyhedra/documentation/old_documentation.m2
+++ b/M2/Macaulay2/packages/Polyhedra/documentation/old_documentation.m2
@@ -1936,8 +1936,7 @@
      
      EXAMPLE {
 	  " P = crossPolytope 2",
-	  " Q = hypercube 2",
-	  " mixedVolume {P,Q}"
+	  " Q = hypercube 2"
 	  }
      
      }
@@ -2232,8 +2231,7 @@
      of a simplex is |det(v_1-v_0,..,v_n-v_0)|/n!, where v_0,..,v_n are the vertices of the simplex.",
      
      EXAMPLE {
-	  " P = crossPolytope 3",
-	  " volume P"
+	  " P = crossPolytope 3"
 	  }
      
      }
@@ -3040,9 +3038,7 @@
      ",TT "to be added",".",
      
      EXAMPLE {
-	  " P = crossPolytope 2",
-	  " Q = secondaryPolytope P",
-	  " vertices Q"
+	  " P = crossPolytope 2"
 	  }
      }
 
--- a/M2/Macaulay2/packages/ToricInvariants.m2
+++ b/M2/Macaulay2/packages/ToricInvariants.m2
@@ -621,14 +621,7 @@
 	    This function computes the polar degrees of the projective toric variety X_A, we do not assume that X_A is normal. The default output is a list of polar degrees; other values of interest computed by the program are also output. To suppress text output use the option Output =>HashTable. 
     	Example
 	    A=matrix{{0, 0, 0, 1, 1,5}, {7,0, 1, 3, 0, -2},{1,1, 1, 1, 1, 1}}
-	    polarDegrees(A)
 	    A=matrix{{3, 0, 0, 1, 1,2},{3,5,0,2,1,3},{0, 1, 2, 0, 2,0},{1, 1, 1, 1, 1,1}}
-	    pdh=polarDegrees(A,Output=>HashTable);
-	    pdh#"polar degrees"
-	    pdh#"dual degree"
-	    pdh#"dual codim"
-	    pdh#"ED"
-	    pdh#"degree"
 Node 
     Key
     	dualDegCodim
@@ -650,10 +643,6 @@
 	    This function computes the degree and codimension of the projective toric variety X_A, we do not assume that X_A is normal. This function uses @TO polarDegrees@ internally and this information can also be obtained from the @TO polarDegrees@ function. 
     	Example
 	    A=matrix{{0, 0, 0, 1, 1,5},{7,0, 1, 3, 0, -2},{1,1, 1, 1, 1, 1}}
-	    dc=dualDegCodim(A)
-	    dc#"dualCodim"
-	    dc#"dualDegree"
-	    pd=polarDegrees(A);
 Node 
     Key
     	cmClass
@@ -677,15 +666,7 @@
 	    This function computes the Chern-Mather class of the projective toric variety X_A pushedforward to the Chow ring of the ambient projective space, we do not assume that X_A is normal. 
     	Example
 	    A=matrix{{0, 0, 0, 1, 1,5},{7,0, 1, 3, 0, -2},{1,1, 1, 1, 1, 1}}
-	    cmClass(A)
 	    A=matrix{{3, 0, 0, 1, 1,2}, {3,5,0,2,1,3},{0, 1, 2, 0, 2,0},{1, 1, 1, 1, 1,1}}
-	    cmh=cmClass(A,Output=>HashTable);
-	    cmh#"CM class"
-	    cmh#"polar degrees"
-	    cmh#"dual degree"
-	    cmh#"dual codim"
-	    cmh#"ED"
-	    cmh#"degree"	    	     	        
 Node 
     Key
     	cmVolumes
@@ -709,9 +690,7 @@
 	    This function computes the Chern-Mather volumes of the projective toric variety X_A, these are the coefficients of the Chern-Mather class ordered from dimension X_A to dimension zero. We do not assume that X_A is normal. 
     	Example
 	    A=matrix{{0, 0, 0, 1, 1,5}, {7,0, 1, 3, 0, -2},{1,1, 1, 1, 1, 1}}
-	    cmVolumes(A)
 	    A=matrix{{3, 0, 0, 1, 1,2}, {3,5,0,2,1,3},{0, 1, 2, 0, 2,0},{1, 1, 1, 1, 1,1}}
-	    cm=cmVolumes(A)
 Node 
     Key
     	edDeg
@@ -735,10 +714,7 @@
 	    This function computes (generic) Euclidean distance degree the projective toric variety X_A, we do not assume that X_A is normal. The default output is a list of polar degrees; other values of interest computed by the program are also output. To suppress text output use the option Output =>HashTable. This function uses @TO polarDegrees@ internally.
     	Example
 	    A=matrix{{0, 0, 0, 1, 1,5}, {7,0, 1, 3, 0, -2},{1,1, 1, 1, 1, 1}}
-	    edDeg(A)
 	    A=matrix{{3, 0, 0, 1, 1,2}, {3,5,0,2,1,3},{0, 1, 2, 0, 2,0},{1, 1, 1, 1, 1,1}}
-	    time edDeg(A)
-	    time edDeg(A,ForceAmat=>true)
 ///
 
 
--- a/M2/configure.ac
+++ b/M2/configure.ac
@@ -958,13 +958,6 @@
      BUILD_nauty=yes
 fi
 
-AC_MSG_CHECKING(whether the package topcom is installed)
-if test "`type -t points2finetrianga`" = file
-then AC_MSG_RESULT(yes)
-else AC_MSG_RESULT([no, will build])
-     BUILD_topcom=yes
-fi
-
 AC_MSG_CHECKING([whether eigen3 library is installed])
 if $PKG_CONFIG --exists eigen3
 then CFO=`$PKG_CONFIG --cflags-only-other eigen3`
