--- a/M2/Macaulay2/packages/=distributed-packages
+++ b/M2/Macaulay2/packages/=distributed-packages
@@ -41,10 +41,8 @@
 EdgeIdeals
 FourTiTwo
 StatePolytope
-Polyhedra
 Truncations
 Polymake
-gfanInterface
 PieriMaps
 Normaliz
 Posets
@@ -176,7 +174,6 @@
 CoincidentRootLoci
 RelativeCanonicalResolution
 RandomCurvesOverVerySmallFiniteFields
-StronglyStableIdeals
 SLnEquivariantMatrices
 CorrespondenceScrolls
 NCAlgebra
