Description: Don't download gftables.
Author: Doug Torrance <dtorrance@piedmont.edu>
Last-Update: 2015-06-04

--- a/M2/configure.ac
+++ b/M2/configure.ac
@@ -1267,7 +1267,7 @@
 test "$LAPACK" = no; val=$?
 AC_DEFINE_UNQUOTED(LAPACK,$val,whether to link with lapack)
 
-BUILD_gftables=yes		dnl it seems that factory now includes gftables, and installs it
+BUILD_gftables=no		dnl it seems that factory now includes gftables, and installs it
 
 SINGULARLIBS="-lfactory  "
 if test $BUILD_factory = no
