Description: Skip building of submodules.
Author: Doug Torrance <dtorrance@monmouthcollege.edu>
Last-Update: 2015-06-04

--- a/M2/GNUmakefile.in
+++ b/M2/GNUmakefile.in
@@ -46,88 +46,88 @@
 	@ echo "-- checking for unstripped binaries"
 	file @pre_programsdir@/* $(BUILTLIBPATH)/bin/* @pre_bindir@/M2-binary usr-dist/bin/M2-binary usr-host/bin/* | (! grep "not stripped")
 
-## submodules
-ifneq (@build_alias@,)
-# it was ../configure --build=@build_alias@
-SUBMODULE_CONFIGOPTIONS += --build=@build_alias@
-endif
-ifneq (@host_alias@,)
-# it was ../configure --build=@host_alias@ (for cross compiling)
-SUBMODULE_CONFIGOPTIONS += --host=@host_alias@
-endif
-SUBMODULES = memtailor mathic mathicgb
-copy-license-files-from-memtailor: LICENSE_FILES = README.md license.txt
-copy-license-files-from-mathic: LICENSE_FILES = README.md lgpl-*
-copy-license-files-from-mathicgb: LICENSE_FILES = README.md gpl-*
-submodules/mathicgb/Makefile: SUBMODULE_CONFIGOPTIONS += --without-tbb
-SUBTARGETS = check all clean distclean run-configure
-GITSUBTARGETS = git-clean git-update git-status git-describe
-all-in-submodules:all-in-libraries
-all-in-mathicgb:all-in-mathic all-in-memtailor
-all-in-mathic:all-in-memtailor
-$(foreach t,$(SUBTARGETS),$(eval $t:$t-in-submodules))
-define submodule-rules
-$(foreach t,$(SUBTARGETS) $(GITSUBTARGETS),$(eval $t-in-submodules:$t-in-$1))
-all-in-$1: run-configure-in-$1 build-in-$1 copy-license-files-from-$1
-copy-license-files-from-submodules: copy-license-files-from-$1
-copy-license-files-from-$1:; $(MKDIR_P) $(BUILTLIBPATH)/licenses/$1 && for i in $$(LICENSE_FILES); do cp -v @srcdir@/submodules/$1/$$$$i $(BUILTLIBPATH)/licenses/$1 ; done
-build-in-$1:; + $(MAKE) -C submodules/$1 all
-check-in-$1: run-configure-in-$1; + ulimit -v 1000000 && $(MAKE) -C submodules/$1 check
-$(foreach c,clean distclean,$(eval $c-in-$1:; if [ -f submodules/$1/Makefile ]; then $(MAKE) -C submodules/$1 $c; fi))
-git-update-in-$1:; cd @srcdir@/submodules/$1 && git checkout master && git pull
-git-status-in-$1:; cd @srcdir@/submodules/$1 && git status
-git-clean-in-$1:; cd @srcdir@/submodules/$1 && git clean -Xdf
-git-describe-in-$1:; cd @srcdir@/submodules/$1 && git describe --dirty --long --always --abbrev=40 --all
-run-configure-in-submodules: run-configure-in-$1
-run-configure-in-$1: submodules/$1/Makefile
-unconfigure-in-submodules: unconfigure-in-$1
-unconfigure-in-$1:; rm -f submodules/$1/Makefile
-submodules/$1/Makefile:@srcdir@/submodules/$1/configure
-	$(MKDIR_P) submodules/$1
-	$(call cdw,submodules/$1) &&							\
-	 @abs_srcdir@/submodules/$1/configure						\
-		--prefix=$(BUILTLIBPATH)						\
-		--with-gtest=yes							\
-		PKG_CONFIG_PATH="$(BUILTLIBPATH)/lib/pkgconfig:$(PKG_CONFIG_PATH)"	\
-		GTEST_PATH=$(BUILTLIBPATH)/include/gtest				\
-		AR=@AR@									\
-		AS=@AS@									\
-		DLLTOOL=@DLLTOOL@							\
-		OBJDUMP=@OBJDUMP@							\
-		STRIP=@STRIP@								\
-		CXXFLAGS="$(CXXFLAGS)"							\
-		CFLAGS="$(CFLAGS)"							\
-		FFLAGS="$(FCFLAGS)"							\
-		FCFLAGS="$(FCFLAGS)"							\
-		TARGET_ARCH=$(TARGET_ARCH)						\
-		CPPFLAGS="$(CPPFLAGS)"							\
-		CC="$(CC)"								\
-		CXX="$(CXX)"								\
-		LDFLAGS="$(LDFLAGS)"							\
-		LOADLIBES="$(LOADLIBES)"						\
-		LDLIBS="$(LDLIBS)"							\
-		$$(SUBMODULE_CONFIGOPTIONS)
-@srcdir@/submodules/$1/configure.ac:
-ifeq (@DOWNLOAD@,yes)
-	cd @srcdir@/.. && git submodule update --init M2/submodules/$1
-else
-	@ echo "error: for the submodule \"$1\"" >&2
-	@ echo "       the source code is not present in the directory \"submodules/$1\"" >&2
-	@ echo "       so either run 'git submodule update --init' in the source tree" >&2
-	@ echo "       or rerun the Macaulay2 \"configure\" command with the added option \"--enable-download\"" >&2
-	@ echo "       to enable automatic downloading of the source code over the internet." >&2
-	@ echo "       To clone a repository together with its submodules, add the option" >&2
-	@ echo "       '--recursive' to the 'git clone' command." >&2
-	@ false
-endif
-@srcdir@/submodules/$1/configure:@srcdir@/submodules/$1/configure.ac @srcdir@/submodules/$1/Makefile.am; cd @srcdir@/submodules/$1 && ./autogen.sh
-endef
-$(foreach s,$(SUBMODULES),$(eval $(call submodule-rules,$s)))
-$(foreach t,all check clean distclean,$(eval $t-in-Macaulay2:$t-in-submodules))
+# ## submodules
+# ifneq (@build_alias@,)
+# # it was ../configure --build=@build_alias@
+# SUBMODULE_CONFIGOPTIONS += --build=@build_alias@
+# endif
+# ifneq (@host_alias@,)
+# # it was ../configure --build=@host_alias@ (for cross compiling)
+# SUBMODULE_CONFIGOPTIONS += --host=@host_alias@
+# endif
+# SUBMODULES = memtailor mathic mathicgb
+# copy-license-files-from-memtailor: LICENSE_FILES = README.md license.txt
+# copy-license-files-from-mathic: LICENSE_FILES = README.md lgpl-*
+# copy-license-files-from-mathicgb: LICENSE_FILES = README.md gpl-*
+# submodules/mathicgb/Makefile: SUBMODULE_CONFIGOPTIONS += --without-tbb
+# SUBTARGETS = check all clean distclean run-configure
+# GITSUBTARGETS = git-clean git-update git-status git-describe
+# all-in-submodules:all-in-libraries
+# all-in-mathicgb:all-in-mathic all-in-memtailor
+# all-in-mathic:all-in-memtailor
+# $(foreach t,$(SUBTARGETS),$(eval $t:$t-in-submodules))
+# define submodule-rules
+# $(foreach t,$(SUBTARGETS) $(GITSUBTARGETS),$(eval $t-in-submodules:$t-in-$1))
+# all-in-$1: run-configure-in-$1 build-in-$1 copy-license-files-from-$1
+# copy-license-files-from-submodules: copy-license-files-from-$1
+# copy-license-files-from-$1:; $(MKDIR_P) $(BUILTLIBPATH)/licenses/$1 && for i in $$(LICENSE_FILES); do cp -v @srcdir@/submodules/$1/$$$$i $(BUILTLIBPATH)/licenses/$1 ; done
+# build-in-$1:; + $(MAKE) -C submodules/$1 all
+# check-in-$1: run-configure-in-$1; + ulimit -v 1000000 && $(MAKE) -C submodules/$1 check
+# $(foreach c,clean distclean,$(eval $c-in-$1:; if [ -f submodules/$1/Makefile ]; then $(MAKE) -C submodules/$1 $c; fi))
+# git-update-in-$1:; cd @srcdir@/submodules/$1 && git checkout master && git pull
+# git-status-in-$1:; cd @srcdir@/submodules/$1 && git status
+# git-clean-in-$1:; cd @srcdir@/submodules/$1 && git clean -Xdf
+# git-describe-in-$1:; cd @srcdir@/submodules/$1 && git describe --dirty --long --always --abbrev=40 --all
+# run-configure-in-submodules: run-configure-in-$1
+# run-configure-in-$1: submodules/$1/Makefile
+# unconfigure-in-submodules: unconfigure-in-$1
+# unconfigure-in-$1:; rm -f submodules/$1/Makefile
+# submodules/$1/Makefile:@srcdir@/submodules/$1/configure
+# 	$(MKDIR_P) submodules/$1
+# 	$(call cdw,submodules/$1) &&							\
+# 	 @abs_srcdir@/submodules/$1/configure						\
+# 		--prefix=$(BUILTLIBPATH)						\
+# 		--with-gtest=yes							\
+# 		PKG_CONFIG_PATH="$(BUILTLIBPATH)/lib/pkgconfig:$(PKG_CONFIG_PATH)"	\
+# 		GTEST_PATH=$(BUILTLIBPATH)/include/gtest				\
+# 		AR=@AR@									\
+# 		AS=@AS@									\
+# 		DLLTOOL=@DLLTOOL@							\
+# 		OBJDUMP=@OBJDUMP@							\
+# 		STRIP=@STRIP@								\
+# 		CXXFLAGS="$(CXXFLAGS)"							\
+# 		CFLAGS="$(CFLAGS)"							\
+# 		FFLAGS="$(FCFLAGS)"							\
+# 		FCFLAGS="$(FCFLAGS)"							\
+# 		TARGET_ARCH=$(TARGET_ARCH)						\
+# 		CPPFLAGS="$(CPPFLAGS)"							\
+# 		CC="$(CC)"								\
+# 		CXX="$(CXX)"								\
+# 		LDFLAGS="$(LDFLAGS)"							\
+# 		LOADLIBES="$(LOADLIBES)"						\
+# 		LDLIBS="$(LDLIBS)"							\
+# 		$$(SUBMODULE_CONFIGOPTIONS)
+# @srcdir@/submodules/$1/configure.ac:
+# ifeq (@DOWNLOAD@,yes)
+# 	cd @srcdir@/.. && git submodule update --init M2/submodules/$1
+# else
+# 	@ echo "error: for the submodule \"$1\"" >&2
+# 	@ echo "       the source code is not present in the directory \"submodules/$1\"" >&2
+# 	@ echo "       so either run 'git submodule update --init' in the source tree" >&2
+# 	@ echo "       or rerun the Macaulay2 \"configure\" command with the added option \"--enable-download\"" >&2
+# 	@ echo "       to enable automatic downloading of the source code over the internet." >&2
+# 	@ echo "       To clone a repository together with its submodules, add the option" >&2
+# 	@ echo "       '--recursive' to the 'git clone' command." >&2
+# 	@ false
+# endif
+# @srcdir@/submodules/$1/configure:@srcdir@/submodules/$1/configure.ac @srcdir@/submodules/$1/Makefile.am; cd @srcdir@/submodules/$1 && ./autogen.sh
+# endef
+# $(foreach s,$(SUBMODULES),$(eval $(call submodule-rules,$s)))
+# $(foreach t,all check clean distclean,$(eval $t-in-Macaulay2:$t-in-submodules))
 
 # installation into usr-dist
 all-in-Macaulay2:install-licenses
-install-licenses:all-in-submodules
+install-licenses:
 	: installing library and submodule licenses in @pre_licensesdir@
 	if [ -d $(BUILTLIBPATH)/licenses/. ] ; \
 	then rm -rf @pre_licensesdir@ ; \
--- a/M2/Macaulay2/bin/Makefile.in
+++ b/M2/Macaulay2/bin/Makefile.in
@@ -182,10 +182,7 @@
 else
 M2_LIBRARIES := $(M2_LIBRARIES) -L../d -lM2inits2
 endif
-M2_LIBDEPS += ../d/libM2inits2.a				\
-	@abs_top_builddir@/submodules/memtailor/libmemtailor.a	\
-	@abs_top_builddir@/submodules/mathic/libmathic.a	\
-	@abs_top_builddir@/submodules/mathicgb/libmathicgb.a
+M2_LIBDEPS += ../d/libM2inits2.a
 ifeq (@CONSTRUCTOR_ORDER_OBJ@,LR)
 M2_OBJECTS := ../d/M2inits.o $(M2_OBJECTS)
 else
