Description: Skip example requiring topcom
Author: Doug Torrance <dtorrance@piedmont.edu>
Last-Update: 2020-05-16

--- a/M2/Macaulay2/packages/Polyhedra/documentation/documentation.m2
+++ b/M2/Macaulay2/packages/Polyhedra/documentation/documentation.m2
@@ -390,8 +390,6 @@
       Text
          This method computes a regular triangulation of a polytope using the package {\tt Topcom.m2}. The output is a list of list of indices which vertices of {\tt P} give a simplex in the triangulation.
 
-      Example
-         T = regularTriangulation hypercube 2
 ///
 
 doc ///
