--- a/M2/Macaulay2/packages/=distributed-packages
+++ b/M2/Macaulay2/packages/=distributed-packages
@@ -41,10 +41,8 @@
 EdgeIdeals
 FourTiTwo
 StatePolytope
-Polyhedra
 Truncations
 Polymake
-gfanInterface
 PieriMaps
 Normaliz
 Posets
@@ -176,15 +174,12 @@
 CoincidentRootLoci
 RelativeCanonicalResolution
 RandomCurvesOverVerySmallFiniteFields
-StronglyStableIdeals
 SLnEquivariantMatrices
 CorrespondenceScrolls
 NCAlgebra
 SpaceCurves
 ExteriorIdeals
-ToricInvariants
 SegreClasses
-SemidefiniteProgramming
 SumsOfSquares
 MultiGradedRationalMap
 VirtualResolutions
