--- a/M2/configure.ac
+++ b/M2/configure.ac
@@ -709,7 +709,7 @@
 #    (Programs linked by this configure script are linked with the options in LIBS.  This allows libraries
 #    dependent on previously detected libraries to be detected by tests that involve linking.
 
-AC_SUBST(LIBLIST, " atomic_ops gc gdbm mpir mpfr readline ntl flint factory lapack mpsolve frobby glpk cddlib fplll givaro linbox boost gtest ")
+AC_SUBST(LIBLIST, " atomic_ops gc gdbm mpir mpfr readline ntl flint factory lapack frobby glpk cddlib fplll givaro linbox boost gtest ")
 # The list LIBLIST is the list of libraries that might be used and linked into M2.
 
 AC_SUBST(PROGLIST, "4ti2 gfan normaliz csdp nauty cddplus lrslib gftables topcom cohomcalg")
@@ -870,13 +870,6 @@
 then BUILTLIBS="-lmpfr $BUILTLIBS"
 fi
 
-# including mpsolve (requires gmp or mpir) 
-AC_LANG(C)
-AC_CHECK_HEADER(mps/mps.h,LIBS="-lmps $LIBS",BUILD_mpsolve=yes)
-if test $BUILD_mpsolve = yes
-then BUILTLIBS="-lmps $BUILTLIBS"
-fi
-
 AC_LANG(C)
 AC_CHECK_HEADER(glpk.h,LIBS="-lglpk $LIBS",BUILD_glpk=yes)
 if test $BUILD_glpk = yes
