Description: Use results of configure -Wsometimes-uninitialized check
Author: Doug Torrance <dtorrance@piedmont.edu>
Last-Update: 2020-05-18

--- a/M2/Macaulay2/d/Makefile.in
+++ b/M2/Macaulay2/d/Makefile.in
@@ -117,8 +117,10 @@
 CXXFLAGS += -Wincompatible-pointer-types-discards-qualifiers
 endif
 else
+ifeq "@HAVE_WSOMETIMES_UNINITIALIZED@" "yes"
 DCFLAGS   += -Wsometimes-uninitialized
 DCXXFLAGS += -Wsometimes-uninitialized
+endif
 # -- non-development --
 DCFLAGS   += -Wno-sometimes-uninitialized
 DCXXFLAGS += -Wno-sometimes-uninitialized
