Description: Copy statuscodes
Origin: https://github.com/Macaulay2/M2/commit/3877132

--- a/M2/Macaulay2/m2/Makefile.in
+++ b/M2/Macaulay2/m2/Makefile.in
@@ -150,7 +150,7 @@
 @pre_packagesdir@ :; @INSTALL@ -d $@
 @pre_packagesdir@/Core  : ; @INSTALL@ -d $@
 @pre_packagesdir@/Core/% : % ; @INSTALL_DATA@ $^ $@
-FILES_TO_COPY := $(DUMPEDM2FILES) loadsequence installedpackages
+FILES_TO_COPY := $(DUMPEDM2FILES) loadsequence installedpackages statuscodes
 all:@pre_packagesdir@ @pre_packagesdir@/Core $(addprefix @pre_packagesdir@/Core/, $(FILES_TO_COPY))
 clean::; rm -rf ../cache/@DUMPDATAFILE@ cache *.obj *.exe runexample temp.data distfiles loads.m2 mpcodes.m2 core gmon.out
 distclean: clean; rm -f Makefile version.m2
