Description: Temporary patch to gather info

--- a/M2/Macaulay2/tests/normal/timing-quotient.m2
+++ b/M2/Macaulay2/tests/normal/timing-quotient.m2
@@ -220,4 +220,6 @@
 d=z^4+z^2*x*y^9+z*x^9*y+x^5*y^5;
 phi=map(P,P,matrix{{x^13*y^4,x^3*y,x^20*y^6*z}});
 tim = timing factor(phi(d));
+tim
+standardSecond
 assert(tim#0 < .05 * standardSecond)
--- a/M2/Macaulay2/tests/Makefile.test.in
+++ b/M2/Macaulay2/tests/Makefile.test.in
@@ -39,7 +39,7 @@
 	   $(notdir $(patsubst %.m2-input, %.out, $(wildcard $(SRCDIR)/*.m2-input)))
 
 PAT := 'internal error|:[0-9][0-9]*:[0-9][0-9]*:\([0-9][0-9]*\):|^GC|^0x|^out of mem|non-zero status|^Command terminated|user.*system.*elapsed|^[0-9]+\.[0-9]+user'
-FILTER := egrep -a $(PAT)
+FILTER := $(if $(VERBOSE),tail,egrep -a $(PAT))
 
 ifeq ($(DEFERRED),no)
 check: $(RESULTS)
