Description: Use Debian KaTeX package for documentation.
Author: Doug Torrance <dtorrance@piedmont.edu>
Forwarded: not-needed
Last-Update: 2020-12-02

--- a/M2/Macaulay2/m2/html.m2
+++ b/M2/Macaulay2/m2/html.m2
@@ -7,9 +7,9 @@
     katexPath := locateCorePackageFileRelative("Style",
 	layout -> replace("PKG", "Style", layout#"package") | "katex", installPrefix, htmlDirectory);
     katexTemplate := ///
-    <link rel="stylesheet" href="%PATH%/katex.min.css" />
-    <script defer="defer" type="text/javascript" src="%PATH%/katex.min.js"></script>
-    <script defer="defer" type="text/javascript" src="%PATH%/contrib/auto-render.min.js"></script>
+    <link rel="stylesheet" href="/usr/share/javascript/katex/katex.css" />
+    <script defer="defer" type="text/javascript" src="/usr/share/javascript/katex/katex.js"></script>
+    <script defer="defer" type="text/javascript" src="/usr/share/javascript/katex/contrib/auto-render.js"></script>
     <script defer="defer" type="text/javascript">
       var macros = {
           "\\break": "\\\\",
@@ -32,8 +32,8 @@
       });
     </script>
     <style type="text/css">.katex { font-size: 1em; }</style>
-    <link href="%PATH%/contrib/copy-tex.min.css" rel="stylesheet" type="text/css" />
-    <script defer="defer" type="text/javascript" src="%PATH%/contrib/copy-tex.min.js"></script>
+    <link href="/usr/share/javascript/katex/contrib/copy-tex.css" rel="stylesheet" type="text/css" />
+    <script defer="defer" type="text/javascript" src="/usr/share/javascript/katex/contrib/copy-tex.js"></script>
     <script defer="defer" type="text/javascript" src="%PATH%/contrib/render-a11y-string.min.js"></script>///;
     LITERAL replace("%PATH%", katexPath, katexTemplate))
 
