Description: Use Debian topcom package
Author: Doug Torrance <dtorrance@piedmont.edu>
Last-Update: 2020-06-22

--- a/M2/configure.ac
+++ b/M2/configure.ac
@@ -986,7 +986,7 @@
 fi
 
 AC_MSG_CHECKING(whether the package topcom is installed)
-if test "`type -t points2finetrianga`" = file
+if test "`type -t topcom-points2finetriangs`" = file
 then AC_MSG_RESULT(yes)
 else AC_MSG_RESULT([no, will build])
      BUILD_topcom=yes
--- a/M2/Macaulay2/packages/Topcom.m2
+++ b/M2/Macaulay2/packages/Topcom.m2
@@ -21,7 +21,7 @@
                 }},
         Headline => "interface to a small part of topcom",
         Configuration => {
-            "path" => ""
+            "path" => "/usr/bin/topcom-"
             },
         PackageImports => {"FourierMotzkin"}
         )
