Description: Use Debian's CSDP package.
Author: Doug Torrance <dtorrance@piedmont.edu>
Last-Update: 2020-05-06

--- a/M2/Macaulay2/packages/SemidefiniteProgramming.m2
+++ b/M2/Macaulay2/packages/SemidefiniteProgramming.m2
@@ -17,7 +17,7 @@
       HomePage => "https://scholar.google.com/citations?user=cFOV7nYAAAAJ&hl=de"}
     },
     Headline => "semidefinite programming",
-    Configuration => {"CSDPexec"=>"","MOSEKexec"=>"mosek","SDPAexec"=>"sdpa","DefaultSolver"=>null},
+    Configuration => {"CSDPexec"=>"/usr/bin/csdp","MOSEKexec"=>"mosek","SDPAexec"=>"sdpa","DefaultSolver"=>null},
     AuxiliaryFiles => true,
     PackageExports => {"NumericalAlgebraicGeometry"}
 )
--- a/M2/configure.ac
+++ b/M2/configure.ac
@@ -931,9 +931,7 @@
 
 AC_MSG_CHECKING(whether the package csdp is installed)
 if test "`type -t csdp`" = file
-then AC_MSG_RESULT([yes, will build anyway])
-     dnl because the package SemidefiniteProgramming doesn't look for it on the PATH
-     BUILD_csdp=yes
+then AC_MSG_RESULT([yes])
 else AC_MSG_RESULT([no, will build])
      BUILD_csdp=yes
 fi
