Description: Use Debian topcom package
Author: Doug Torrance <dtorrance@piedmont.edu>
Last-Update: 2020-06-22

--- a/M2/configure.ac
+++ b/M2/configure.ac
@@ -1021,10 +1021,8 @@
 fi
 
 AC_MSG_CHECKING(whether the package topcom is installed)
-if test "`type -t points2finetriangs`" = file
-then AC_MSG_RESULT([yes, will build anyway])
-     BUILD_topcom=yes
-     dnl currently the package Topcom uses only the internal Macaulay2 path to run its programs
+if test "`type -t topcom-points2finetriangs`" = file
+then AC_MSG_RESULT([yes])
 else AC_MSG_RESULT([no, will build])
      BUILD_topcom=yes
 fi
--- a/M2/Macaulay2/packages/Topcom.m2
+++ b/M2/Macaulay2/packages/Topcom.m2
@@ -21,7 +21,7 @@
                 }},
         Headline => "interface to a small part of topcom",
         Configuration => {
-            "path" => ""
+            "path" => "/usr/bin/topcom-"
             },
         PackageImports => {"FourierMotzkin"}
         )
