Description: Use system gc
Author: Doug Torrance <dtorrance@debian.org>
Last-Update: 2022-04-19

--- a/M2/configure.ac
+++ b/M2/configure.ac
@@ -783,11 +783,6 @@
      fi
 fi
 
-## we always build gc so we can configure it for large memory spaces
-## see issue #2445
-BUILD_gc=yes
-BUILD_ALWAYS="$BUILD_ALWAYS gc"
-
 AC_MSG_CHECKING(whether package bdw-gc is provided)
 dnl Note: we don't include -lgccpp below, because linking with that library is what replaces ::new and friends with GC_MALLOC_UNCOLLECTABLE and friends:
 if [ pkg-config --exists bdw-gc ]
