Description: Use Debian's CSDP package.
Author: Doug Torrance <dtorrance@piedmont.edu>
Last-Update: 2020-05-06

--- a/M2/Macaulay2/packages/SemidefiniteProgramming.m2
+++ b/M2/Macaulay2/packages/SemidefiniteProgramming.m2
@@ -17,7 +17,7 @@
       HomePage => "https://scholar.google.com/citations?user=cFOV7nYAAAAJ&hl=de"}
     },
     Headline => "semidefinite programming",
-    Configuration => {"CSDPexec"=>"","MOSEKexec"=>"mosek","SDPAexec"=>"sdpa","DefaultSolver"=>null},
+    Configuration => {"CSDPexec"=>"/usr/bin/csdp","MOSEKexec"=>"mosek","SDPAexec"=>"sdpa","DefaultSolver"=>null},
     AuxiliaryFiles => true,
     PackageImports => {"SimpleDoc"},
     PackageExports => {"NumericalAlgebraicGeometry"}
