Description: Add Verbose option for check
Author: Doug Torrance <dtorrance@piedmont.edu>
Last-Update: 2020-10-31

--- a/M2/Macaulay2/m2/testing.m2
+++ b/M2/Macaulay2/m2/testing.m2
@@ -39,16 +39,9 @@
      runString(teststring, pkg, usermode);
      )
 
-check = method(Options => {UserMode => null})
-check String  := opts -> pkg -> check(needsPackage (pkg, LoadDocumentation => true), opts)
-check Package := opts -> pkg -> (
-    if not pkg.Options.OptionalComponentsPresent then (
-        stderr << "--warning: optional components required for " <<
-            toString pkg << " tests are not present; skipping" << endl;
-        return);
-    pkg = prep pkg;
-    scan(keys pkg#"test inputs", n -> onecheck(n, pkg, if opts.UserMode === null then not noinitfile else opts.UserMode));
-    if hadError then error(toString numErrors, " error(s) occurred running tests for package ", toString pkg);)
+check = method(Options => {UserMode => null, Verbose => false})
+check String  := opts -> pkg -> check(-1, pkg, opts)
+check Package := opts -> pkg -> check(-1, pkg, opts)
 
 check(ZZ, String)  := opts -> (n, pkg) -> check(n, needsPackage (pkg, LoadDocumentation => true), opts)
 check(ZZ, Package) := opts -> (n, pkg) -> (
@@ -57,5 +50,20 @@
             toString pkg << " tests are not present; skipping" << endl;
         return);
     pkg = prep pkg;
-    onecheck(n, pkg, if opts.UserMode === null then not noinitfile else opts.UserMode);
-    if hadError then error("test #", toString n, " of package ", toString pkg, " failed");)
+    errorList := {};
+    scan(if n == -1 then keys pkg#"test inputs" else {n}, k -> (
+            previousNumErrors := numErrors;
+            onecheck(k, pkg, if opts.UserMode === null then not noinitfile else opts.UserMode);
+            if numErrors > previousNumErrors then
+                errorList = append(errorList, k)));
+    if hadError then error("test", if numErrors > 1 then "s" else "",
+        " #", demark(", ", toString \ errorList),
+        " of package ", toString pkg, " failed",
+        if opts.Verbose then (
+            numTests := if n == -1 then pkg#"test number" else 1;
+            ":" | newline |
+            concatenate(apply(errorList, k ->
+                newline | get("!tail " | temporaryDirectory() |
+                toString(temporaryFilenameCounter + 2 * (k - numTests)) |
+                ".tmp")))
+        ) else "");)
--- a/M2/Macaulay2/packages/Macaulay2Doc/functions/check-doc.m2
+++ b/M2/Macaulay2/packages/Macaulay2Doc/functions/check-doc.m2
@@ -50,7 +50,7 @@
     pkg:Package
       or String, the package to test
     i:ZZ
-      the index of the test to run
+      the index of the test to run or -1 to run all tests
     UserMode=>Boolean
       if true, do not use the @TT "-q"@ in arguments to the Macaulay2 executable when running tests,
       thereby allowing it to load the user's @TO "initialization file"@, allowing it to load packages
@@ -58,6 +58,8 @@
       and allowing packages it loads to read their configuration files from the user's
       @TO2 {"applicationDirectory", "application directory"}@. If false, the @TT "-q"@ argument is added.
       If @TO "null"@, then add @TT "-q"@ if it appears as an option in @TO "commandLine"@.
+    Verbose=>Boolean
+      if true, then print the output of all failing tests
   Consequences
     Item
       the tests in the package @TT "pkg"@ are run (in separate Macaulay2 processes, with the random number
--- a/M2/Macaulay2/packages/Makefile.in
+++ b/M2/Macaulay2/packages/Makefile.in
@@ -41,7 +41,7 @@
 	$(eval check::check-$i)\
 	$(eval check-$i:; \
 		if ! grep "CacheExampleOutput => true" @srcdir@/$i.m2 >/dev/null ;\
-		then @pre_bindir@/M2 -q --no-preload $(STOP) -e "needsPackage(\"$i\",LoadDocumentation=>true,DebuggingMode=>true); check($i,UserMode=>false); exit 0" ;\
+		then @pre_bindir@/M2 -q --no-preload $(STOP) -e "needsPackage(\"$i\",LoadDocumentation=>true,DebuggingMode=>true); check($i,UserMode=>false,Verbose=>$(Verbose)); exit 0" ;\
 		fi ))
 info-dir: @pre_infodir@/dir
 @pre_infodir@/dir:|@pre_infodir@
